pkg/kubectl/cmd/create.go

```diff
@@ -172,3 +181,54 @@ func createAndRefresh(info *resource.Info) error {
 	info.Refresh(obj, true)
 	return nil
 }
+
+// nameFromCommandArgs is a utility function for commands that assume the first argument is a resource name
+func nameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) == 0 {
+		return "", cmdutil.UsageError(cmd, "NAME is required")
+	}
+	return args[0], nil
+}
+
+// createSubcommandInternal is intended to be used by subcommands that generator client side objects
+func createSubcommandInternal(f *cmdutil.Factory, cmd *cobra.Command, name string, out io.Writer, structuredGenerator kubectl.StructuredGenerator) error {
```

Comment:
> I assume you also need this to be a public exposed func?

I'm not fully convinced that we'll implement our create subcommands in this way, but if we decide to, I'd want to be able to use this.